kratos/python_scripts/multistage_orchestrators/multistage_orchestrator.py

```diff
@@ -0,0 +1,180 @@
+import sys
+import importlib
+
+import KratosMultiphysics
+from KratosMultiphysics.project import Project
+from KratosMultiphysics.model_parameters_factory import KratosModelParametersFactory
+
+class MultistageOrchestrator():
+
+    def __init__(self, project : Project) -> None:
+        """Base class for multistage orchestrators
+        
+        Member variables:
+        project -- Kratos project instance
+        """
+
+        # Store pointer to current project
+        # Note that the project already contains the multistage simulation settings
+        self.__project = project
+
+        # Add Kratos version and compilation to settings
+        kratos_version = f"{KratosMultiphysics.KratosGlobals.Kernel.Version()}-{KratosMultiphysics.KratosGlobals.Kernel.BuildType()}"
+        self.__project.GetSettings().AddString("kratos_version", kratos_version)
+
+    def Run(self):
+        """Main function that runs the complete multistage simulation."""
+
+        err_msg = "Calling base MultistageOrchestrator Run() method. This must be implemented in derived orchestrators."
+        raise NotImplementedError (err_msg)
+
+    def CheckStageSettings(self, stage_name : str):
+        """Check the settings for the given stage name
+        
+        This methods performs the check of the given stage name settings.
+        Note that the Check() method of the stage instance is intentionally not called in here.
+        This is due to the fact that the Check() of the stage instance is to be called within its Initialize() method.
+        """
+
+        if self.__project.GetSettings()["stages"][stage_name].Has("stage_postprocess"):
+            if self.__project.GetSettings()["stages"][stage_name]["stage_postprocess"].Has("modelers"):
+                err_msg = f"Found 'modelers' field in 'stage_postprocess' of stage {stage_name}."
+                err_msg += " Place the 'modelers' section in the next stage 'stage_preprocess'."
+                raise Exception(err_msg)
+
+    def CreateStage(self, stage_name : str):
+        """This method creates a stage instance
+
+        Given a stage name, this method creates the corresponding stage instance from the "analysis_stage" registry entry in the settings.
+        The stage instance is to be created from the registry. If not registered, it tries to use "analysis_stage" as Python module.
+        """
+
```

Comment:
Agree. In the short term I think we should move this piece of code to a stage factory module. I note that this can be used in both the single stage and multistage contexts.

Nonetheless, I'd leave it for a future PR.

Reply:
So would you please ad a TODO comment to avoid more copy paste of it?

Reply:
Sure.

Reply:
Done.